Library/Homebrew/livecheck/version.rb

```diff
@@ -0,0 +1,69 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    # A formula or cask version, split into its component sub-versions.
+    #
+    # @api private
+    class LivecheckVersion
+      extend T::Sig
+
+      include Comparable
+
+      sig { params(formula_or_cask: T.any(Formula, Cask::Cask), version: Version).returns(LivecheckVersion) }
+      def self.create(formula_or_cask, version)
+        versions = case formula_or_cask
+        when Formula
+          [version]
+        when Cask::Cask
+          version.to_s.split(/[,:]/).map { |s| Version.new(s) }
```

Comment:
Not blocking but if there's a chance for tests for any of this that might be nice?

Reply:
[We’re considering getting rid of `:` in `version`](https://github.com/Homebrew/homebrew-cask/issues/95207). Pointing that out so there’s something to link to from the other issue, and we can revisit this code when the time comes.